From fa77c4362848c897961ed853f3dfc35d665254de Mon Sep 17 00:00:00 2001
From: Sekhar Nori <nsekhar@ti.com>
Date: Thu, 28 Jun 2012 20:29:10 +0530
Subject: [PATCH 8/9] ARM: OMAP2+: AM335x: hwmod: do not reset GPIO0

Do not reset GPIO0 at  boot-up because GPIO0 is used
on AM335x EVM-SK to control VTT regulators on DDR3.

Without this some EVM-SK boards fail to boot-up because
of DDR3 corruption.

Ideally this change should be done only for EVM-SK, but
hwmod data is common across all boards and and there is
no known side-effect of not resetting GPIO0 at boot.

Suggested-by: Vaibhav Bedia <vaibhav.bedia@ti.com>
Signed-off-by: Sekhar Nori <nsekhar@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index de7a3ab..5a4660b 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -949,7 +949,7 @@ static struct omap_hwmod am33xx_gpio0_hwmod = {
 	.name		= "gpio1",
 	.class		= &am33xx_gpio_hwmod_class,
 	.clkdm_name	= "l4_wkup_clkdm",
-	.flags		= HWMOD_CONTROL_OPT_CLKS_IN_RESET,
+	.flags		= (HWMOD_CONTROL_OPT_CLKS_IN_RESET | HWMOD_INIT_NO_RESET),
 	.mpu_irqs	= am33xx_gpio0_irqs,
 	.main_clk	= "dpll_core_m4_div2_ck",
 	.prcm		= {
-- 
1.7.7.6

